### DISCUSSION TITLE
Add buttons to header or body component - datetimepicker

### DISCUSSION THREAD

**Discussion by EdierBernal** [matheo/angular#63] [Q&A]
Hi, Thanks for all, your code is very usefull :) 

I'm trying to add some buttons to set days on calendar but I don't have idea how to edit the headder or body component. 

Do you have some idea about that?
![Screenshot 2022-08-01 164000](https://user-images.githubusercontent.com/23252048/182251655-7ec1c083-43d1-4461-afb7-2e81c5c15c7f.png)

**Comment by matheo** [comment]
@EdierBernal sorry the late answer, but still I'm afraid that you need a lot of familiarity with the internals of the component to figure out the way to inject your own header.

`<mat-datepicker>` can receive a `calendarHeaderComponent`: https://github.com/matheo/angular/blob/master/libs/datepicker/src/lib/datepicker-base.ts#L304
but I don't know if you can inject your own actions/buttons in the footer.

You can customize the mat-header component
https://github.com/matheo/angular/blob/master/libs/datepicker/src/lib/mat-header.html
and put the actions there maybe.
